pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,121 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+# pylint: disable=too-many-branches
+def contract_tensors(
+    tensors: Sequence,
+    communication_graph: MultiDiGraph,
+    prepare_nodes: Sequence[Sequence[PrepareNode]],
+    measure_nodes: Sequence[Sequence[MeasureNode]],
+    use_opt_einsum: bool = False,
+):
+    """Contract tensors according to the edges specified in the communication graph.
+
+    This operation is differentiable. The ``prepare_nodes`` and ``measure_nodes`` arguments are both
+    sequences of size ``len(communication_graph.nodes)`` that describe the order of indices in the
+    ``tensors`` with respect to to the :class:`~.PrepareNode` and :class`~.MeasureNode` edges in the
+    communication graph.
+
+    Args:
+        tensors (Sequence): the tensors to be contracted
+        communication_graph (MultiDiGraph): the communication graph determining connectivity between
+            the tensors
+        prepare_nodes (Sequence[PrepareNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of preparation indices in each tensor
+        measure_nodes (Sequence[MeasureNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of measurement indices in each tensor
+        use_opt_einsum (bool): Determines whether to use the
+            [opt_einsum](https://dgasmith.github.io/opt_einsum/) package. This package is useful for
+            tensor contractions of large networks but must be installed separately using, e.g.,
+            ``pip install opt_einsum``.
+
+    Returns:
+        float or array-like: the result of contracting the tensor network
+
+    **Example**
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
```

Comment:
Oh wait, so this is not meant to be user-facing? I would move this `note` right to the top of the docstring in that case :) Otherwise a user that chances upon it will spend time trying to understand it rather than being pointed to `qcut`!

Reply:
Compare the docstrings for [`np.sum`](https://numpy.org/doc/stable/reference/generated/numpy.sum.html) (user facing) with [`ndarray.sum()`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.sum.html) (not user facing)

Reply:
> Oh wait, so this is not meant to be user-facing?

In most cases a user wouldn't interact directly with this function. The intention is for this function to be used within a `processing_fn` that takes a flat list of results and returns the result of the original uncut circuit (to be added in the next PR). So users will either use cut circuit (high level, uses `processing_fn` under the hood) or work directly with `processing_fn` (for more advanced users). However, I didn't want to make this `contract_tensors` entirely missing from the docs.

Reply:
> I would move this note right to the top of the docstring in that case

Done!